libs/inform/src/lib/services/modal/modal.service.ts

```diff
@@ -44,25 +49,23 @@ export class NgxModalService {
 	public open<ActionType extends string = string, DataType = any>(
 		options: NgxModalOptions<ActionType, DataType>
 	): Observable<ActionType> {
-		// Iben: If there still is an active subject running, the modal is still active and we early exit.
-		if (this.modalClosedSubject !== undefined && !this.modalClosedSubject?.closed) {
+		// Iben: If a previous modal is still active, we early exit.
+		if (this.hasModalSubject.value) {
```

Comment:
Not a big fan of the `.value` syntax because it gets the current value synchronously from an inherently asynchronous stream. Not telling you to not do this if this is the best way to go about it, but keep in mind that this can cause unwanted side effects. Only use the `.value` property when there is no other way to go about it.

Reply:
Sidenote: wouldn't it be better if the options had a `hasPriority` value or something that can indicate if the new modal should close the previous one? I can imagine that your second modal getting "lost" because of an other one still being open, might be unwanted behaviour.

Reply:
Whilst I understand your concerns with the value call here, in this case you just want to know the current value and not respond every time the value updates. A subscription here would mean you'd place a `take(1)`, and at that point it becomes the same as requesting the current value with the value call. I would even go as far as to say that the value approach is better in that case, because you don't start a subscription when it's not needed.

When it comes to the second remark, is that something other modal packages also do? 🤔 AFAIK all packages I've ever used just don't allow you to add a second modal if there's already one open.

Reply:
I think it would just be a nice to have to allow for certain modals to have a higher priority and thus closing a previous one. Maybe a nice to have feature for the board, but something to think about.